### PR TITLE
fall back to non-fcu fork choice on epoch boundaries

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: beacon_chain/consensus_object_pools/consensus_manager.nim [hunk 1]
```diff
@@ -340,11 +340,11 @@ proc getGasLimit*(
 from ../spec/datatypes/bellatrix import PayloadID
 
 proc runProposalForkchoiceUpdated*(
-    self: ref ConsensusManager, wallSlot: Slot) {.async.} =
+    self: ref ConsensusManager, wallSlot: Slot): Future[bool] {.async.} =
```
Comment (etan-status) [inline beacon_chain/consensus_object_pools/consensus_manager.nim]:
```suggestion
    self: ref ConsensusManager, wallSlot: Slot): Future[Opt[void]] {.async.} =
```

File: beacon_chain/consensus_object_pools/consensus_manager.nim [hunk 2]
```diff
@@ -340,11 +340,11 @@ proc getGasLimit*(
...
   let
     nextWallSlot = wallSlot + 1
     (validatorIndex, nextProposer) = self.checkNextProposer(wallSlot).valueOr:
-      return
+      return false
```
Comment (etan-status) [inline beacon_chain/consensus_object_pools/consensus_manager.nim]: `? self.checkNextProposer(wallSlot)`, if going with `Opt[void]`

**Review by etan-status — APPROVED** [review]

File: beacon_chain/consensus_object_pools/consensus_manager.nim (hunk 2 above)
Comment (tersec) [reply beacon_chain/consensus_object_pools/consensus_manager.nim]: Yeah, will update these to be consistent

Comment (etan-status) [reply beacon_chain/consensus_object_pools/consensus_manager.nim]: also, Future[Opt[void]], accidentally forgot the Future wrapper on first attempt

Comment (tersec) [reply beacon_chain/consensus_object_pools/consensus_manager.nim]: https://github.com/status-im/nimbus-eth2/pull/5195/commits/88ea3e23303d1d4fa833a5012a48583bdb5aad09

I did try `? self.checkNextProposer` and it didn't seem to work, but can investigate later. The `valueOr` approach still works.

**Pull request merged**